src/screens/watch/watch.tsx

```diff
@@ -29,6 +35,14 @@ export class WatchScreenComponent extends React.Component<
         Clipboard.setString(getApnsToken());
     };
 
+    public onQrCodeScanned = async (value: string) => {
+        WalletConnectClient.connect(value);
+        const keychainPassword = await getPassword();
+        if (keychainPassword) {
+            storeEncrypted(value, WC_CONNECTION_STRING, keychainPassword.password);
```

Comment:
i think this is not quite ok, we need to store the session not the connection string. I think that if we try to reconnect with connection string and the other party is not connected the handshake cannot be done. If we store the session we just connect back, no handshake is required.